lib/vmfloaty/pooler.rb

```diff
@@ -54,25 +63,29 @@ def self.retrieve(verbose, os_type, token, url)
     end
   end
 
-  def self.modify(verbose, url, hostname, token, lifetime, tags)
+  def self.modify(verbose, url, hostname, token, modify_hash)
     if token.nil?
       raise TokenError, "Token provided was nil. Request cannot be made to modify vm"
     end
 
-    modify_body = {}
-    if lifetime
-      modify_body['lifetime'] = lifetime
-    end
-    if tags
-      modify_body['tags'] = tags
+    modify_hash.each do |key, value|
+      unless [:tags, :lifetime, :disk].include? key
+        STDERR.puts "Warning: configured service type does not support modification of #{key}."
```

Comment:
I updated this to use an exception instead, but I think the message is still necessary; nspooler supports modification of a "reason for reservation", which isn't a thing in vmpooler; There's a (slim) chance someone may try to set a reason to a vmpooler instance, so this is meant to guard against that.

Reply:
Ahhh okay cool 👍 Yeah the main reason why I like having these as exceptions instead of printing is because the `Pooler` class is also meant to be consumed as a library, and if a print happens the library consume can't control that. But yeah, I'm all for guarding against something like this in the future!